core/rfb.js

```diff
@@ -315,8 +321,18 @@ export default class RFB extends EventTargetMixin {
             this._rfbConnectionState === "connected") {
             if (viewOnly) {
                 this._keyboard.ungrab();
+                (async () => {
+                    if (await isAsyncClipboardAvailable()) {
+                        this._clipboard.ungrab();
```

Comment:
Can't we avoid this fiddling with event handlers by simply checking our state in the focus handler? Would be a lot more neatly contained in a single place.